src/meta/src/hummock/manager/mod.rs

```diff
@@ -850,6 +853,36 @@ where
                 )?;
             }
             versioning.current_version = new_version;
+
+            let worker_nodes = self
+                .cluster_manager
+                .list_worker_node(WorkerType::ComputeNode, None)
+                .await;
+            let mut global_min = *versioning
+                .hummock_version_deltas
+                .last_key_value()
+                .unwrap()
+                .0;
+            for worker_node in worker_nodes {
+                if let Some(min_pinned_id) = versioning
+                    .pinned_versions
+                    .get(&worker_node.id)
+                    .map(|pinned_version| pinned_version.min_pinned_id)
+                {
+                    if min_pinned_id < global_min {
+                        global_min = min_pinned_id;
+                    }
+                    host2pin.push((WorkerKey(worker_node.host.unwrap()), min_pinned_id));
+                }
+            }
+            deltas_to_send = versioning
+                .hummock_version_deltas
+                .range((Excluded(global_min), Unbounded))
+                .map(|(k, v)| (*k, v.clone()))
+                .collect();
+            self.metrics
+                .delta_payload_size
```

Comment:
What about 2 metrics:
1. size as [encoded_len](https://github.com/singularity-data/risingwave/blob/579669235d5588aad6c6a80f9c5d3a9232dbdb30/src/meta/src/hummock/metrics_utils.rs#L35)? As payload may varies greatly in real size.
2. len as current deltas_to_send.len(). This can reveal if there is any version pinned for a long time.

Reply:
> What about 2 metrics:
> 
> 1. size as [encoded_len](https://github.com/singularity-data/risingwave/blob/579669235d5588aad6c6a80f9c5d3a9232dbdb30/src/meta/src/hummock/metrics_utils.rs#L35)? As payload may varies greatly in real size.
> 2. len as current deltas_to_send.len(). This can reveal if there is any version pinned for a long time.

fixed